js/glowingbear.js

```diff
@@ -687,7 +687,26 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         }
     };
 
+    
+    $scope.parseHash = function() {
+
+        //Fill in url parameters, they take precedence over the stored settings, but store them
+        var params = $location.$$hash.split('&').map(function(val) { return {key: val.split('=')[0], value: val.split('=')[1]}; });
```

Comment:
according to the linter there's a semicolon missing on this line in column 126 but tbh it's so dense I can't really parse it

Reply:
In JS, mappers can have side effects, so you can do
```js
var params;
$location.$$hash.split('&').map(function(val) {
    var segs = val.split('=');
    params[segs[0]] = segs[1];
});
```
and then you can simply use `params['host']` etc

Reply:
I initially went for a find() with a lambda but this is not ES6.

Reply:
Or even dot notation params.host, the linter suggested.